eng/test-job.yml

```diff
@@ -12,364 +12,42 @@ parameters:
   corefxTests: false
   displayNameArgs: ''
   runInUnloadableContext: false
-  publishTestArtifacts: false
+  publishTestArtifacts: true
```

Comment:
I believe the attempt to have the default value of false was to save space.

To anyone reviewing this in the future, now the run-test job depends on this publishing so we will always need to publish tests.

Reply:
Hmm, I guess that probably also means we no longer need this parameter ;-). After all, from the telemetry viewpoint, I think that always producing a "test drop" is general goodness.

Reply:
I agree that it is worth removing.

Reply:
As of the latest version of this PR, the parameter 'publishTestArtifacts' exists no more.